### PR TITLE
Add gas measurements for VRF requests and fulfillments

### PR DESCRIPTION
**Pull request by coventry**
Total round-trip cost is about 380k gas. ~175514 gas for the request, and ~195654 for the fulfillment.

### CODE REVIEW

File: core/services/vrf/vrf_fulfillment_cost_test.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	proofBlob, err := proof.MarshalForSolidityVerifier()
+	require.NoError(t, err, "could not marshal VRF proof for VRFCoordinator!")
+	rawData, err := coordinator.coordinatorABI.Pack("fulfillRandomnessRequest",
+		proofBlob[:])
+	require.NoError(t, err)
+	callMsg := ethereum.CallMsg{From: coordinator.neil.From,
+		To: &coordinator.rootContractAddress, Data: rawData}
+	estimate, err := coordinator.backend.EstimateGas(context.TODO(), callMsg)
+	require.NoError(t, err)
+	assert.Less(t, estimate, uint64(200000))
```
Comment (kaleofduty) [inline core/services/vrf/vrf_fulfillment_cost_test.go]: maybe add an assert.More with a lower bound to ensure that the we have a tight estimate?

File: core/services/vrf/vrf_request_cost_test.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+func TestMeasureRandomnessRequestGasCost(t *testing.T) {
+	coordinator := deployCoordinator(t)
+	keyHash_, _, fee := registerProvingKey(t, coordinator)
+	rawData, err := coordinator.consumerABI.Pack("requestRandomness",
+		common.BytesToHash(keyHash_[:]), fee, one)
+	require.NoError(t, err)
+	callMsg := ethereum.CallMsg{To: &coordinator.consumerContractAddress, Data: rawData}
+	estimate, err := coordinator.backend.EstimateGas(context.TODO(), callMsg)
+	require.NoError(t, err)
+	assert.Less(t, estimate, uint64(180000))
```
Comment (kaleofduty) [inline core/services/vrf/vrf_request_cost_test.go]: maybe add an assert.More with a lower bound to ensure that the we have a tight estimate?

**Comment by coventry** [comment]
@lorenzb , thanks for the feedback. There seems to be great variation in the fulfillment gas cost, from 148k to 195k. Request gas cost is a constant 175k, in my hands.

File: core/services/vrf/vrf_fulfillment_cost_test.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import (
+	"context"
+	"testing"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMeasureFulfillmenttGasCost(t *testing.T) {
```
Comment (j16r) [inline core/services/vrf/vrf_fulfillment_cost_test.go]: This test style is quite hard to read, would you consider chinking blocks into logical segments? I sort of see three logical groupings here: setup, execution and assertions.

Comment (coventry) [reply core/services/vrf/vrf_fulfillment_cost_test.go]: Thanks for the feedback. I factor the gas-cost test into a utility function, and broke up the new gas-cost test functions as you suggested.

**Pull request merged**